### DISCUSSION TITLE
Using Astro, Starlight Theme and AnalogJs do not play together well

### DISCUSSION THREAD

**Discussion by bjornharvold** [analogjs/analog#1278] [General]
Hi @brandonroberts 

When you have some free time, can you please have a quick look at the discussion that's been going on on the Starlight theme repo here:

https://github.com/withastro/starlight/discussions/2198

In particular, have a look at the example repo I point at there.

In a nutshell:

- I add Astro to my Nx monorepo using https://github.com/nxtensions/nxtensions/blob/main/packages/astro/README.md
- I added Starlight theme
- I added AnalogJS

I run `nx dev docs`. I get no errors when not using Starlight. The starlight guys get no errors when removing AnalogJS.

Your input would be much appreciated 🙇‍♂️
Bjorn

**Comment by adrian-ub** [comment]
Hi @bjornharvold , how were you able to solve it?

I'm not using Nx but I get the same error and I already created the `tsconfig.app.json` file

**Reply by brandonroberts** [reply]
What's the error? Do you have a branch?

**Reply by adrian-ub** [reply]
Repo: https://github.com/adrian-ub/astro-angular

![image](https://github.com/user-attachments/assets/a9b57ded-3d6f-44c1-afc0-ab53b6768639)

When moving to `/index-2` it works correctly (without starlight), the solution for the moment is to disable `expressiveCode`

**Reply by adrian-ub** [reply]
This error only occurs in dev, when doing the build it works correctly

**Reply by brandonroberts** [reply]
Try updating the analog integration to this

```ts
    analogjsangular({
      vite: {
        transformFilter(_code, id) {
          return id.includes('src/components')
        }
      }
    }),
```

Then delete the `.astro` directory and run `npm run dev` again

**Reply by adrian-ub** [reply]
amazing, it works

console.log result:

![image](https://github.com/user-attachments/assets/9eab42d6-1250-4ee9-bc72-7fb19d69a1bf)

![image](https://github.com/user-attachments/assets/36badef6-dfdf-417f-a0fb-531662cfc126)

**Reply by brandonroberts** [reply]
Cool! I thought we had this covered in our docs

**Reply by brandonroberts** [reply]
Added some docs

https://analogjs.org/docs/packages/astro-angular/overview#filtering-file-transforms

**Comment by Suoerman108** [comment]
ok
